Plumbox Launcher and CLI

Talina06 opened the pull request.

Talina06 committed:
Plumbox Launcher and CLI

Talina06 committed:
Rearranged code

Talina06 committed:
Rearranged code

meson10 commented on src/main/java/in/oogway/plumbox/cli/Localstorage.java:
```diff
@@ -0,0 +1,40 @@
+import java.util.UUID;
+
+public class Localstorage extends StorageDriver {
+    YamlHandler yamlHandler = YamlHandler.createYamlHandler();
```
The format is JSON now right? It cannot be YAML. If the Local implementation is not used, please delete it.

meson10 commented on src/main/java/in/oogway/plumbox/cli/Plumbox.java:
```diff
@@ -0,0 +1,143 @@
+        }
+
+        if (this.filesystem.equals("LOCAL")) {
+            this.storageDriver = new Localstorage();
```
LocalStorage stores as YAML whereas Redis sores JSON. Keep them as same or drop the one that is not being used.

meson10 commented on src/main/resources/input_source_file.json:
```diff
@@ -0,0 +1 @@
+[{"busID":1,"boxID":1,"latitude":18.531806,"longitude":73.850043,"timestamp":1516749989},{"busID":1,"boxID":1,"latitude":18.541184,"longitude":73.826923,"timestamp":1516748789},{"busID":1,"boxID":1,"latitude":18.531541,"longitude":73.849947,"timestamp":1516747349},{"busID":1,"boxID":1,"latitude":18.531541,"longitude":73.849947,"timestamp":1516745909},{"busID":1,"boxID":1,"latitude":18.541184,"lon ...
```
This should be in src/test and not in main.

meson10 replied:
Also, this doesn't belong to this project.

meson10 commented on src/main/java/in/oogway/plumbox/launcher/transformer/Transformer.java:
```diff
@@ -1,5 +1,8 @@
+
 public interface Transformer {
-    void run();
+    Dataset<Row> run(Dataset<Row> events);
```
rows and not events.

meson10 commented on src/main/java/in/oogway/plumbox/launcher/storage/RedisStorage.java:
```diff
@@ -1,15 +1,20 @@
+
+    default void redisServer() {
+        String address = Config.getDirPath("redis_server_address");
+        Config.jedis = new Jedis(address);
```
Not sure if this deserves to go to Config. It should be a static/private singleton.

meson10 commented on src/main/java/in/oogway/plumbox/launcher/storage/JSONParser.java:
```diff
@@ -0,0 +1,38 @@
+*   @author talina06 on 2/15/18
+*/
+
+public interface JSONParser extends RedisStorage{
```
JSONParser is too ambiguous a name.

meson10 commented on src/main/java/in/oogway/plumbox/launcher/storage/JSONParser.java:
```diff
@@ -0,0 +1,38 @@
+    default Map loadContent(String id)
+            throws IllegalAccessException, InstantiationException, ClassNotFoundException, IOException {
+        byte[] bytes = read(id);
+        Reader fileReader = getFileReader(bytes);
```
I am not sure if converting to file and loading is the best way to load content from JSON.

meson10 commented on src/main/java/in/oogway/plumbox/launcher/source/Source.java:
```diff
@@ -0,0 +1,34 @@
+import java.io.Reader;
+import java.util.Map;
+
+public class Source implements JSONParser{
```
Use some generic here. implements JSONParser < KnownSourceSchema >

Talina06 committed:
Refactoring code

meson10 committed:
Work Draft v1

meson10 committed:
Rename Transformation with Pipeline

meson10 committed:
Rename Piepline.class

Talina06 committed:
Minor Tweaks

Talina06 committed:
Merge branch 'work_one' of https://github.com/oogway/plumboxlauncher into work_one

meson10 committed:
plumbox can add generic sources

 - Use -P java style properties.
 - Make CLI dumber. Doesnt validate arguments as of now.
 - Sample MySQL source in tests.
 - gradle.properties with readonly credentials for Plumbox.

meson10 committed:
Resolved conflicts

Talina06 committed:
uncomment AWS S3 read task

meson10 committed:
Merge pull request #5 from oogway/work_one

Work Draft v1

Talina06 commented:
Work_branch already merged.
